octokit/search.go

```diff
@@ -2,47 +2,60 @@ package octokit
 
 import (
 	"github.com/jingweno/go-sawyer/hypermedia"
-	"net/url"
 )
 
-var SearchURL = Hyperlink("search{/type}?q={query}{&page,per_page,sort,order}")
+var SearchURITemplate = "search{/type}?q={query}{&page,per_page,sort,order}"
 
-func (c *Client) Search(url *url.URL) (searches *SearchService) {
-	searches = &SearchService{client: c, URL: url}
-	return
+func (c *Client) Search(uriTemplate string) *SearchService {
+	return &SearchService{&GenericService{client: c, uriTemplate: uriTemplate}}
 }
 
 // A service to return search records
 type SearchService struct {
-	client *Client
-	URL    *url.URL
+	*GenericService
 }
 
 // Get the user search results based on SearchService#URL
-func (g *SearchService) Users() (userSearchResults UserSearchResults,
+func (g *SearchService) Users(params M) (userSearchResults UserSearchResults,
```

Comment:
Instead of explicitly setting a key to `nil` I am checking if the key `uri` is present at all in the `params` map.